frappe/core/doctype/doctype/doctype.json

```diff
@@ -517,12 +524,51 @@
    "fieldname": "email_settings_sb",
    "fieldtype": "Section Break",
    "label": "Email Settings"
+  },
+  {
+   "depends_on": "eval:!doc.istable && !doc.issingle",
+   "fieldname": "sharing_rules_section",
+   "fieldtype": "Section Break",
+   "label": "Sharing Rules"
+  },
+  {
+   "default": "0",
```

Comment:
Default should be 1. Also not sure if there is a use case for disabling it. Maybe we only need the fields `allow_write_for_assigned_users` and `allow_share_for_assigned_users`.